applications/luci-app-wireguard/luasrc/view/wireguard.htm

```diff
@@ -169,100 +79,117 @@
 
   XHR.poll(5, '<%=REQUEST_URI%>', { status: 1 },
    function(x, data) {
-    for (var i = 0, ilen = data.length; i < ilen; i++) {
-      var iface = data[i];
-      var ifid = iface['public_key'] + "_";
-      var s = String.format(
-        '<strong><%:Public Key%>: </strong>%s' +
-        '<br /><strong><%:Listening Port%>: </strong>%<%:s%>',
-        iface['public_key'],
-        iface['listening_port']
-      );
-      document.getElementById(ifid + "info").innerHTML = s;
-      for (var j = 0, jlen = iface['peers'].length; j < jlen; j++) {
-        var peer = iface['peers'][j];
-        var pid = ifid + peer['public_key'] + "_";
-        s = String.format(
+    for (var key in data) {
+      if (!data.hasOwnProperty(key)) { continue; }
+      var ifname = key;
+      var iface = data[key];
+      var s = "";
+      if (iface.public_key == '(none)') {
+        s += '<em><%:Interface not operational!%></em>';
+      } else {
+        s += String.format(
           '<strong><%:Public Key%>: </strong>%s',
-          peer['public_key']
+          iface.public_key
         );
-        if (peer['endpoint'] != '(none)') {
+      }
+      if (iface.listen_port > 0) {
+        s += String.format(
+          '<br /><strong><%:Listen Port%>: </strong>%<%:s%>',
+          iface.listen_port
+        );
+      }
+      if (iface.fwmark != 'off') {
+        s += String.format(
+          '<br /><strong><%:Firewall Mark%>: </strong>%<%:s%>',
+          iface.fwmark
+        );
+      }
+      document.getElementById(ifname + "_info").innerHTML = s;
+      for (var i = 0, ilen = iface.peers.length; i < ilen; i++) {
+        var peer = iface.peers[i];
+        if (peer.public_key == '(none)') {
```

Comment:
This situation will never occur. Peers will always have public keys. You can thus get rid of this conditional.